src/nupic/regions/SDRClassifierRegion.py

```diff
@@ -0,0 +1,502 @@
+
+    self.recordNum += 1
+
+
+  def customCompute(self, recordNum, patternNZ, classification):
+    """
+    Just return the inference value from one input sample. The actual 
+    learning happens in compute() -- if, and only if learning is enabled -- 
+    which is called when you run the network.
+    
+    WARNING: The method customCompute() is here to maintain backward 
+    compatibility. This method is deprecated, and will be removed.
+    Use network.run() instead, which will call the compute() method.
+
+    Parameters:
+    --------------------------------------------------------------------
+    recordNum:      Record number of the input sample.
+    patternNZ:      List of the active indices from the output below
+    classification: Dict of the classification information:
+                      bucketIdx: index of the encoder bucket
+                      actValue:  actual value going into the encoder
+
+    retval:     dict containing inference results, one entry for each step in
+                self.steps. The key is the number of steps, the value is an
+                array containing the relative likelihood for each bucketIdx
+                starting from bucketIdx 0.
+
+                for example:
+                  {'actualValues': [0.0, 1.0, 2.0, 3.0]
+                    1 : [0.1, 0.3, 0.2, 0.7]
+                    4 : [0.2, 0.4, 0.3, 0.5]}
+    """
+
+    # If the compute flag has not been initialized (for example if we 
+    # restored a model from an old checkpoint) initialize it to False.
+    if not hasattr(self, "_computeFlag"):
+      self._computeFlag = False
+
+    if self._computeFlag:
+      # Will raise an exception if the deprecated method customCompute() is 
+      # being used at the same time as the compute function.
+      warnings.simplefilter('error', DeprecationWarning)
+      warnings.warn("The customCompute() method should not be "
+                    "called at the same time as the compute() "
+                    "method. The compute() method is called "
+                    "whenever network.run() is called.",
+                    DeprecationWarning)
+
+    return self._sdrClassifier.compute(recordNum,
+                                       patternNZ,
+                                       classification,
+                                       self.learningMode,
+                                       self.inferenceMode)
+
+
+  def getOutputValues(self, outputName):
+    """
+    Return the dictionary of output values. Note that these are normal Python
+    lists, rather than numpy arrays. This is to support lists with mixed scalars
+    and strings, as in the case of records with categorical variables
+    """
+    return self._outputValues[outputName]
+
+
+  def getOutputElementCount(self, outputName):
+    """Returns the width of dataOut."""
+
+    # Check if classifier has a 'maxCategoryCount' attribute
+    if not hasattr(self, "maxCategoryCount"):
```

Comment:
This if statement is not required - this region will always have maxCategoryCount because it is initialized in the constructor.